src/lib/operators/aggregate_hash.cpp

```diff
@@ -519,12 +581,24 @@ void AggregateHash::_aggregate() {
             results.resize(1);
             results[0].aggregate_count += input_chunk_size;
           } else {
-            // count occurrences for each group key
-            for (ChunkOffset chunk_offset{0}; chunk_offset < input_chunk_size; chunk_offset++) {
-              auto& result = get_or_add_result(result_ids, results,
-                                               get_aggregate_key<AggregateKey>(keys_per_chunk, chunk_id, chunk_offset),
-                                               RowID{chunk_id, chunk_offset});
-              ++result.aggregate_count;
+            // Count occurrences for each group key -  If we have more than one aggregate function (and thus more than
+            // one context), it makes sense to cache the results indexes, see get_or_add_result for details.
+            if (_contexts_per_column.size() > 1) {
```

Comment:
Can you think of a way to avoid repeating the for loop?

Reply:
Which for loop?

Reply:
587 and 595, depending on `_contexts_per_column.size()`.

Reply:
Argh. Wrong branch :D

I guess we could wrap it into another lambda, but it would not make it easier to understand. Moving the `if` into the loop would cause the conditional to be evaluated more than once.

Reply:
That's what I thought about as well but I hoped you have another idea. :D

Reply:
So far, I haven't seen a nice way to promote run-time to compile-time bools in c++ :/